core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -1903,13 +1903,20 @@ private[spark] class DAGScheduler(
 
           case smt: ShuffleMapTask =>
             val shuffleStage = stage.asInstanceOf[ShuffleMapStage]
-            shuffleStage.pendingPartitions -= task.partitionId
```

Comment:
for deterministic stages, we do accept laggard tasks?

Reply:
Yes, if determinate, this is fine.